src/ChainSafe.GamingSDK.EVM/Web3/Core/Web3.cs

```diff
@@ -51,29 +52,29 @@ private static T AssertComponentAccessible<T>(T value, string propertyName)
             return value;
         }
 
-        void IDisposable.Dispose()
+        async ValueTask IAsyncDisposable.DisposeAsync()
         {
-            Terminate();
+            await TerminateAsync();
             GC.SuppressFinalize(this);
         }
 
-        public async ValueTask Initialize()
+        public async ValueTask InitializeAsync()
         {
             if (initialized)
             {
                 throw new Web3Exception("Web3 was already initialized.");
             }
 
-            if (provider != null)
+            foreach (var lifecycleParticipant in serviceProvider.GetServices<ILifecycleParticipant>())
```

Comment:
This can be a separate service, just to keep web3 cleaner. This way web3 could be treated as a facade delegating all of the work to its components.

Reply:
Lifecycle participation is already its own service. If you mean the part where it's calling the implementations, it really can't be a separate service since `Web3` would have to call that other service anyway, there's simply no other place to be calling the code from. It'd be a layer of indirection that doesn't really do anything.

Reply:
You know, I kind of feel we're over-engineering here. Too much code in one class is bad, but too many classes with few lines of actual logic is also bad for readability. Let's keep this, and when it does make sense to have a separate service, we'll add it in.